metricshub-hardware/src/main/java/org/metricshub/hardware/util/MonitorNameBuilder.java

```diff
@@ -0,0 +1,1020 @@
+	 *
+	 * @param monitor the {@link Monitor} instance representing the network card device
+	 * @return the formatted network card name
+	 */
+	public static String buildNetworkCardName(final Monitor monitor) {
+		// Check the attributes
+		final Map<String, String> attributes = monitor.getAttributes();
+		Assert.notNull(attributes, ATTRIBUTES_CANNOT_BE_NULL);
+
+		// Network card vendor without unwanted words
+		String networkCardVendor = attributes.get(VENDOR);
+		if (hasMeaningfulContent(networkCardVendor)) {
+			networkCardVendor = NETWORK_VENDOR_MODEL_TRIM_PATTERN.matcher(networkCardVendor).replaceAll("");
+		}
+
+		// Network card model without unwanted words and up to 30 characters
+		String networkCardModel = attributes.get(MODEL);
+		if (hasMeaningfulContent(networkCardModel)) {
+			networkCardModel = NETWORK_VENDOR_MODEL_TRIM_PATTERN.matcher(networkCardModel).replaceAll("");
+			if (networkCardModel.length() > 30) {
+				networkCardModel = networkCardModel.substring(0, 30);
+			}
+		}
+
+		// Build the name
+		return buildName(
+			null,
+			attributes.get(DISPLAY_ID),
+			attributes.get(DEVICE_ID),
+			attributes.get(ID_COUNT),
+			NETWORK_CARD_TRIM_PATTERN,
+			attributes.get(NETWORK_DEVICE_TYPE),
+			networkCardVendor,
+			networkCardModel
+		);
+	}
+
+	/**
+	 * Builds the other device name based on the current implementation.
+	 *
+	 * @param monitor the {@link Monitor} instance representing the other device
+	 * @return the formatted device name
+	 */
+	public static String buildOtherDeviceName(final Monitor monitor) {
+		// Check the attributes
+		final Map<String, String> attributes = monitor.getAttributes();
+		Assert.notNull(attributes, ATTRIBUTES_CANNOT_BE_NULL);
+
+		// Build the name
+		return buildName(
+			attributes.get(OTHER_DEVICE_TYPE), //#question
+			attributes.get(DISPLAY_ID),
+			attributes.get(DEVICE_ID),
+			attributes.get(ID_COUNT),
+			OTHER_DEVICE_TRIM_PATTERN,
+			attributes.get(ADDITIONAL_LABEL)
+		);
+	}
+
+	/**
+	 * Builds the physical disk name based on the current implementation.
+	 *
+	 * @param monitor the {@link Monitor} instance representing the physical disk device
+	 * @return the formatted physical disk name
+	 */
+	public static String buildPhysicalDiskName(final Monitor monitor) {
+		// Check the attributes
+		final Map<String, String> attributes = monitor.getAttributes();
+		Assert.notNull(attributes, ATTRIBUTES_CANNOT_BE_NULL);
+
+		// Retrieve the size metric
+		final Map<String, AbstractMetric> metrics = monitor.getMetrics();
+		final AbstractMetric diskSize = metrics.get(PHYSICAL_DISK_SIZE_METRIC);
+
+		// Build the name
+		return buildName(
+			attributes.get(PHYSICAL_DISK_DEVICE_TYPE),
+			attributes.get(DISPLAY_ID),
+			attributes.get(DEVICE_ID),
+			attributes.get(ID_COUNT),
+			PHYSICAL_DISK_TRIM_PATTERN,
+			attributes.get(VENDOR),
+			humanReadableByteCountSI(diskSize != null ? Double.toString(diskSize.getValue()) : null)

```

Comment:
Converting the value to `String` explicitly isn't necessary. Keep it as a `Double` and update `humanReadableByteCountSI` to accept a  `Double` value.